packages/juce-blueprint/src/lib/BlueprintBackend.ts

```diff
@@ -157,19 +145,116 @@ export default {
     __rootViewInstance = new ViewInstance(id, 'View');
 
     return __rootViewInstance;
-  },
+}
+
+function __bubbleEvent(view: Instance, eventType: string, event: any): void {
+    while (view && view !== __getRootContainer() && event.bubbles) {
+        if (view instanceof ViewInstance) {
+          const hasHandler = view._props.hasOwnProperty(eventType) &&
+            typeof view._props[eventType] === 'function';
+
+          if (hasHandler) {
+            view._props[eventType](event);
+          }
+        }
 
+      view = view._parent;
+    }
+}
+
+//TODO: Consider more performant LCA algorithms.
+//      Could try something like: https://www.geeksforgeeks.org/lca-n-ary-tree-constant-query-o1
+//      though this would require a level of book-keeping in appendChild, removeChild etc.
+//      Lets just get the behaviour correct and tested first.
+function __getLCA(root: Instance, lhs: Instance, rhs: Instance): Instance | null {
+    if (root == null)
+        return null;
+
+    // If lhs and rhs are the same component self is the lowest common ancestor.
+    if (lhs.getViewId() === rhs.getViewId())
+        return lhs;
+
+    if (root.getViewId() === lhs.getViewId() || root.getViewId() === rhs.getViewId())
+        return root;
+
+    let i = 0;
+    let lastLCA: Instance | null = null;
+
+    // RawTextViewInstance does not hold a child array so
+    // treat it as empty here.
+    if (!(root instanceof RawTextViewInstance)) {
```

Comment:
This could just be a check on whether _children is undefined but I think it's better to be explicit here for the moment.